### PR TITLE
new Directive for addPermission

### PR DESCRIPTION
**Pull request by scherfm**
The EmptyRequestId is changed to a random one because i wanted to see if any changes from me in the package org.elasticmq.rest.sqs has an effect, they have but adding a new rest function/directive doesn't compile because of type not found errors.

This code isn't working at the moment, the PR is just to show my changes.

### CODE REVIEW

File: rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SQSRestServerBuilder.scala [hunk 1]
```diff
@@ -141,7 +142,8 @@ case class TheSQSRestServerBuilder(providedActorSystem: Option[ActorSystem],
         changeMessageVisibilityBatch(p) ~
         deleteQueue(p) ~
         getQueueAttributes(p) ~
-        setQueueAttributes(p)
+        setQueueAttributes(p) ~
+        addPermissions(p)
```
Comment (adamw) [inline rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SQSRestServerBuilder.scala]: I think there's a type here: you defined the method as `addPermission` but here you call `addPermissions`

Comment (scherfm) [inline rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SQSRestServerBuilder.scala]: you are totally right about this but even if i change this to addPermission the errors stay the same:
> [error] /home/marcel/dev/sources/elasticmq-server/rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SQSRestServerBuilder.scala:116: not found: type AddPermissionDirectives
> [error]       with AddPermissionDirectives
> [error]            ^
> [error] /home/marcel/dev/sources/elasticmq-server/rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SQSRestServerBuilder.scala:146: not found: value addPermission
> [error]         addPermission(p)
> [error]         ^
> [error] two errors found
> [error] (elasticmq-rest-sqs/compile:compileIncremental) Compilation failed

Comment (adamw) [inline rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SQSRestServerBuilder.scala]: Ah! you are missing a `.scala` file extension in `AddPermissionDirectives` :)

Comment (scherfm) [inline rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SQSRestServerBuilder.scala]: That's it! Thank you very much! IntelliJ hides the file extensions in the project tree so i didn't see that there was none, now everything compiles just fine and i can finally start to work on the function. Thanks again.

Comment (adamw) [inline rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/SQSRestServerBuilder.scala]: Great. If you could update the PR, I'll release a minor version

**Comment by scherfm** [comment]
Updated the PR. Tested my changes with the amazon-aws-api php lib.
For clarification a short summarize of the change:
If an addPermission request is made the queue server now always answers with an api-accepted response xml instead of a 404 - Not found Error.

```
<AddPermissionResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/">
          <ResponseMetadata>
            <RequestId>15639547-7534-1569-8246-865798464538</RequestId>
          </ResponseMetadata>
</AddPermissionResponse>
```

**Comment by adamw** [comment]
Released 0.13.1 - thanks!

**Pull request merged**